bundles/org.openhab.voice.watsonstt/src/main/java/org/openhab/voice/watsonstt/internal/WatsonSTTService.java

```diff
@@ -157,16 +159,17 @@ public STTServiceHandle recognize(STTListener sttListener, AudioStream audioStre
         return new STTServiceHandle() {
             @Override
             public void abort() {
-                var socket = socketRef.get();
-                if (socket != null) {
-                    socket.close(1000, null);
-                    socket.cancel();
-                    try {
-                        Thread.sleep(100);
-                    } catch (InterruptedException ignored) {
+                if (!aborted.getAndSet(true)) {
+                    var socket = socketRef.get();
```

Comment:
Is there no risk to not close the socket ?
This would be the case when onDisconnected is first called. You are setting aborted to true. Then when abort is called, you will not enter in the if.

Reply:
It shouldn't as the sdk executes onDisconnect after the socket is closed, so if it was executed the socket is already closed, so no needed to close it again. 
On the error method I'm not setting abort to true because I can not asume there that the socket is closed.

Reply:
Ok